From 96d62f21f52e543dfb1f114988d4912ef321772f Mon Sep 17 00:00:00 2001
From: Jakob Petsovits <jpetso@petsovits.com>
Date: Thu, 4 Apr 2024 19:03:49 -0400
Subject: [PATCH 1/2] kcms/mouse: Remove unused xorg-evdev dependency

It's not necessary to access it directly anymore, now that we only
expose settings for libinput.

(cherry picked from commit b4fe3e9115856cdd8a67f11d1ab9b5246e2c4770)
---
 kcms/mouse/CMakeLists.txt     | 3 ---
 kcms/mouse/backends/x11.cmake | 1 -
 kcms/mouse/inputbackend.h     | 1 -
 3 files changed, 5 deletions(-)

diff --git a/kcms/mouse/CMakeLists.txt b/kcms/mouse/CMakeLists.txt
index 639981380..909ed815d 100644
--- a/kcms/mouse/CMakeLists.txt
+++ b/kcms/mouse/CMakeLists.txt
@@ -32,9 +32,6 @@ if (BUILD_KCM_MOUSE_KWIN_WAYLAND)
 endif()
 
 if (BUILD_KCM_MOUSE_X11)
-    if (NOT EVDEV_FOUND)
-        list(APPEND MISSING_DEPS "xorg-evdev")
-    endif()
     if (NOT XORGLIBINPUT_FOUND)
         list(APPEND MISSING_DEPS "xorg-libinput")
     endif()
diff --git a/kcms/mouse/backends/x11.cmake b/kcms/mouse/backends/x11.cmake
index 3b1601285..c402612de 100644
--- a/kcms/mouse/backends/x11.cmake
+++ b/kcms/mouse/backends/x11.cmake
@@ -11,7 +11,6 @@ set(backend_SRCS
 set(backend_LIBS
     ${backend_LIBS}
     PkgConfig::XORGLIBINPUT
-    PkgConfig::EVDEV
     X11::X11
     X11::Xi
     X11::Xcursor
diff --git a/kcms/mouse/inputbackend.h b/kcms/mouse/inputbackend.h
index 579634d27..0a6e6ac6a 100644
--- a/kcms/mouse/inputbackend.h
+++ b/kcms/mouse/inputbackend.h
@@ -20,7 +20,6 @@ enum class InputBackendMode {
 #endif
 #if BUILD_KCM_MOUSE_X11
     XLibinput = 1,
-    XEvdev = 2,
 #endif
 };
 
-- 
2.45.1


From fcc965d6a90822b2c6600bf2426dfb0c2b46021a Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Mon, 20 May 2024 18:41:57 +0200
Subject: [PATCH 2/2] Cleanup remaining pkg_check_modules(EVDEV) call

Amends 181b6c85ea6d46652b136462d5b6e0c9b56fc5c2 and b4fe3e9115856cdd8a67f11d1ab9b5246e2c4770

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
(cherry picked from commit 2242b7852f1ab746e4f6929e2fcd51b24b2ec37e)
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b475d3c5b..a515367a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -226,7 +226,6 @@ add_feature_info("XCB-XKB" XCB_XKB_FOUND "Required for building kcm/keyboard")
 add_feature_info("libxft" X11_Xft_FOUND "X FreeType interface library required for font installation")
 
 pkg_check_modules(XORGLIBINPUT xorg-libinput IMPORTED_TARGET)
-pkg_check_modules(EVDEV xorg-evdev>=2.8.99.1 IMPORTED_TARGET)
 pkg_check_modules(XORGSERVER xorg-server IMPORTED_TARGET)
 
 ecm_find_qmlmodule(org.kde.pipewire 0.1)
-- 
2.45.1

